### PR TITLE
Error log display enhancement for link cluster dialog

### PR DESCRIPTION
**Pull request by t-rufang**
- Updated console view
error log with color_update2

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java [hunk 1]
```diff
@@ -137,10 +160,34 @@ protected void textChanged(DocumentEvent e) {
         getOKAction().setEnabled(false);
     }
 
+    protected void printLogLine(@NotNull ConsoleViewContentType logLevel, @NotNull String log) {
+        consoleViewPanel.print(LocalDateTime.now().toString() + " " + logLevel.toString().toUpperCase() + " " + log + "\n", logLevel);
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: Could you use `jodatime` to replace `LocalDateTime`? the jodatime can provide time zone with date.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java [hunk 2]
```diff
@@ -215,7 +262,7 @@ protected boolean isBasicAuthSelected() {
         return ((String) authComboBox.getSelectedItem()).equalsIgnoreCase("Basic Authentication");
     }
 
-    protected void basicValidate() {
+    protected void basicValidation() {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: Let the method name be `doSomething` patten:
```suggestion
    protected void validateBasicInputs() {
```

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/serverexplore/AddNewClusterCtrlProvider.java [hunk 3]
```diff
@@ -51,6 +55,12 @@
             "Please click'OK',then accept the untrusted certificate \r\n"+
             "if you want to link to this cluster.Or you can update the \r\n" +
             "host to link to a different cluster.";
+    private static final String InvalidCertificateErrorMsg =
+            "ERROR: The cluster certificate is invalid.";
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/serverexplore/AddNewClusterCtrlProvider.java]: Is `ERROR` duplicated with log level?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java [hunk 4]
```diff
@@ -78,6 +87,10 @@
     protected HintTextField arisHostField;
     protected HintTextField arisClusterNameField;
     protected JPanel arisLivyServiceCard;
+    protected HideableDecorator errorMessageDecorator;
+    private JPanel errorMessagePanelHolder;
+    private JPanel errorMessagePanel;
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: It looks like a message panel, not only an error message panel

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java (hunk 1 above)
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: And could you combine this function with the one of provision dialog?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java [hunk 5]
```diff
@@ -137,10 +160,34 @@ protected void textChanged(DocumentEvent e) {
...
+
+
     // Data -> Components
     @Override
     public void setData(@NotNull AddNewClusterModel data) {
-        errorMessageField.setText(data.getErrorMessage());
+        // clear console view panel before set new error message
+        consoleViewPanel.clear();
+
+        if (data.getErrorMessageList() != null && data.getErrorMessageList().size() > 0) {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: Are you going the combine the error message list with history messages?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java (hunk 1 above)
Comment (t-rufang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/serverexplore/ui/AddNewClusterForm.java]: Replaced with `DateTime` in joda time. E.g. `2019-01-15T14:57:53.587+08:00`

---
In reply to: [247745657](https://github.com/Microsoft/azure-tools-for-java/pull/2663#discussion_r247745657) [](ancestors = 247745657)

**Comment by t-rufang** [comment]
Feature request: #2668

**Pull request merged**